pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/DistinctCountThetaSketchAggregationFunction.java

```diff
@@ -576,6 +616,69 @@ private SetOperationBuilder getSetOperationBuilder() {
         : SetOperation.builder().setNominalEntries(_thetaSketchParams.getNominalEntries());
   }
 
+  /**
+   * Validates that the function context's substitution parameters ($1, $2, etc) does not exceed the number
+   * of predicates passed into the post-aggregation function.
+   *
+   * For example, if the post aggregation function is:
+   * INTERSECT($1, $2, $3)
+   *
+   * But there are only 2 arguments passed into the aggregation function, throw an error
+   * @param context The parsed function context that's a tree structure
+   * @param numPredicates Max number of predicates available to be substituted
+   */
+  private static void validatePostAggregationExpression(final ExpressionContext context, final int numPredicates) {
+    if (context.getType() == ExpressionContext.Type.LITERAL) {
+      throw new IllegalArgumentException("Invalid post-aggregation function expression syntax.");
+    }
+
+    if (context.getType() == ExpressionContext.Type.IDENTIFIER) {
+      int id = extractSubstitutionPosition(context.getIdentifier());
+      if (id <= 0)
+        throw new IllegalArgumentException("Argument substitution starts at $1");
+      if (id > numPredicates)
+        throw new IllegalArgumentException("Argument substitution exceeded number of predicates");
+      // if none of the invalid conditions are met above, exit out early
+      return;
+    }
+
+    if (!MergeFunction.isValid(context.getFunction().getFunctionName())) {
+      final String allowed =
+          Arrays.stream(MergeFunction.values())
+              .map(MergeFunction::name)
+              .collect(Collectors.joining(","));
+      throw new IllegalArgumentException(
+          String.format("Invalid Theta Sketch aggregation function. Allowed: [%s]", allowed));
+    }
+
+    switch (MergeFunction.valueOf(context.getFunction().getFunctionName().toUpperCase())) {
+      case SET_DIFF:
+        // set diff can only have 2 arguments
+        if (context.getFunction().getArguments().size() != 2) {
+          throw new IllegalArgumentException("SET_DIFF function can only have 2 arguments.");
+        }
+        break;
+      case SET_UNION:
+      case SET_INTERSECT:
```

Comment:
Check it has more than one argument?
Prevent inefficient function such as `UNION()` or `UNION($1)`

Reply:
Done